src/System/IO/BlockIO/URingFFI.hsc

```diff
@@ -26,6 +29,42 @@ foreign import capi unsafe "liburing.h io_uring_queue_init"
 foreign import capi unsafe "liburing.h io_uring_queue_exit"
   io_uring_queue_exit :: Ptr URing -> IO ()
 
+foreign import capi safe "liburing.h io_uring_queue_init_params"
+  io_uring_queue_init_params :: CUInt -> Ptr URing -> Ptr URingParams -> IO CInt
+
+iORING_SETUP_CQSIZE :: CUInt
+iORING_SETUP_CQSIZE = #{const IORING_SETUP_CQSIZE}
+
+data {-# CTYPE "liburing.h" "struct io_uring_params" #-}
+     URingParams = URingParams {
+                      sq_entries :: !CUInt,
+                      cq_entries :: !CUInt,
+                      flags      :: !CUInt,
+                      features   :: !CUInt
+                      -- Note: this is a subset of all the fields. These are
+                      -- just the ones we need now or are likely too need.
+                      -- If you need more, just add them.
+                    }
+  deriving stock (Show, Eq)
+
+instance Storable URingParams where
+  sizeOf    _    = #{size      struct io_uring_params}
+  alignment _    = #{alignment struct io_uring_params}
+  peek      p    = do sq_entries     <- #{peek struct io_uring_params, sq_entries} p
+                      cq_entries     <- #{peek struct io_uring_params, cq_entries} p
+                      flags          <- #{peek struct io_uring_params, flags} p
+                      features       <- #{peek struct io_uring_params, features} p
+                      return URingParams {..}
+  poke      p ps = do -- As we only cover a subset of the fields, we must clear
+                      -- the remaining fields we don't set to avoid them
+                      -- containing arbitrary values.
+                      fillBytes p 0 #{size struct io_uring_params}
```

Comment:
:+1: I was already wondering whether we were not leaving some of the memory uninitialised, seeing as the `man` page for `io_uring_setup` says: "The resv array must be initialized to zero."